src/unstract/sdk/file_storage/fs_interface.py

```diff
@@ -0,0 +1,61 @@
+from abc import ABC, abstractmethod
+from typing import Union
+
+from fsspec import AbstractFileSystem
+
+from unstract.sdk.file_storage.constants import Common, FileSeekPosition
+
+
+class FileStorageInterface(ABC):
+    @abstractmethod
+    def open(
+        self,
+        path: str,
+        mode: str,
+        encoding: str = Common.DEFAULT_ENCODING,
+    ) -> Union[AbstractFileSystem]:
+        pass
+
+    def read(
```

Comment:
@gaya3-zipstack Won't we need `abstractmethod` decorator here too?

Reply:
Yes I missed this. Added now. Thanks!

Reply:
@gaya3-zipstack is the latest code not pushed. Don't see the decorator here.

Reply:
Yea, I was just addressing and replying here for tracking. Pushed latest a bit back.